File: client/src/lib/invoke/searchFirstUsers.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { Config } from "@bindings";

export function searchFirstUsers(config: Config, search: string): ResultAsync<string[], string> {
    return ResultAsync.fromPromise(
        invoke("search_first_users", { config, search }),
        (error) => `Failed to search first users: ${error}`
    );
}

File: client/src-tauri/src/commands/search_first_users.rs
use crate::config::Config;
use nexium::gitlab::GitlabClient;

#[tauri::command]
pub async fn search_first_users(
    config: Config,
    search: String,
) -> Result<Vec<String>, String> {
    tauri::async_runtime::spawn_blocking(move || {
        GitlabClient::new(config.gitlab_token, config.gitlab_token_type)
            .search_users(&search)
            .map_err(|e| e.to_string())
    })
    .await
    .map_err(|e| format!("Failed to search first users: {}", e))?
}
